compression/compression.ml

```diff
@@ -0,0 +1,89 @@
+module type ALGORITHM = sig
+  val executable : string
+end
+
+open Xapi_stdext_pervasives.Pervasiveext
+
+module Make(Algorithm : ALGORITHM) = struct
+  let available () = Sys.file_exists Algorithm.executable
+
+  (** Helper function to prevent double-closes of file descriptors *)
+  let close to_close fd =
+    if List.mem fd !to_close then Unix.close fd;
```

Comment:
Close could raise `EBADF` (that should not happen), `EINTR` (this may happen), `EIO` (this may happen). Should we protect this close in a `try ... with` block to make sure that all the cleanup is performed?